src/Http/Middleware/CsrfProtectionMiddleware.php

```diff
@@ -240,12 +304,18 @@ public function createToken(): string
      */
     protected function _verifyToken(string $token): bool
     {
-        if (strlen($token) <= self::TOKEN_VALUE_LENGTH) {
+        $decoded = base64_decode($token);
+        // If base64 fails we're in a compatibility mode from before
+        // tokens were salted on each request.
+        if ($decoded === false) {
+            $decoded = $token;
```

Comment:
Should we add a warning here? Users shouldn't see this occur more than once per request/refresh, yes?

Reply:
I don't think we should show explicit warnings for internal changes which have been mitigated and don't require the dev to make any changes.

Reply:
What I am asking is whether this introduces a security risk with the fallback.

Reply:
I don't think there is anything new being introduced. Without this flow we wouldn't be able to accept old tokens which would break applications as they upgrade.

Reply:
Ok. As long as they ensure the token is refreshed. Up to them. We should probably test this though since codecov says it's untested.